<?xml version="1.0"?>
<launch>

  <arg name="id_robot" default="robot"/>
  <arg name="docker_action_server" default="accurate_docking/pp_docker"/>
  <arg name="move_action_server" default="accurate_docking/move"/>
  <arg name="base_frame" default="$(arg id_robot)_base_link"/>
  <arg name="pregoal_frame" default="$(arg id_robot)_accurate_test_goal"/>
  <arg name="goal_frame" default="$(arg id_robot)_accurate_test_goal"/>
  <arg name="reflectors_frame" default="$(arg id_robot)_filtered_point_between_reflectors"/>
  <arg name="pregoal_offset_1" default="[-0.05, 0.0, 0.0]"/>
  <arg name="pregoal_offset_2" default="[-0.05, 0.0, 0.0]"/>
  <arg name="step_back_distance" default="1.3"/>
  <arg name="consecutive_iterations" default="5"/>
  <arg name="skip_pregoal_2" default="false"/>
  
  <group ns="$(arg id_robot)">

      <node name="accurate_docking" pkg="accurate_docking" type="accurate_docking_node.py" output="screen">
        <param name="docker_action_server" value="$(arg docker_action_server)"/>
        <param name="move_action_server" value="$(arg move_action_server)"/>
        <param name="base_frame" value="$(arg base_frame)"/>
        <param name="pregoal_frame" value="$(arg pregoal_frame)"/>
        <param name="goal_frame" value="$(arg goal_frame)"/>
        <param name="reflectors_frame" value="$(arg reflectors_frame)"/>
        <param name="pregoal_offset_1" value="$(arg pregoal_offset_1)"/>
        <param name="pregoal_offset_2" value="$(arg pregoal_offset_2)"/>
        <param name="step_back_distance" value="$(arg step_back_distance)"/>
        <param name="consecutive_iterations" value="$(arg consecutive_iterations)"/>
        <param name="skip_pregoal_2" value="$(arg skip_pregoal_2)"/>
      </node>

  </group>


</launch>
